applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -145,28 +144,57 @@ static void handle_dvfs_error(int32_t err)
 	(void) k_work_reschedule(&dvfs_retry.retry_work, timeout);
 }
 
-static void set_dvfs_freq(enum dvfs_frequency_setting target_freq)
+static void dvfs_notify_cb(void)
+{
+	int res;
+	int32_t ret = sys_notify_fetch_result(&cli.notify, &res);
+
+	if (ret < 0) {
+		LOG_DBG("work not completed");
+		return;
+	}
+
+	ret = nrf_clock_control_release(dvfs_clock_dev, &spec);
+	if (ret < 0) {
+		LOG_ERR("Failed to release requested clock specs");
+		module_set_state(MODULE_STATE_ERROR);
+		module_state = STATE_ERROR;
+		cancel_dvfs_retry_work();
+		return;
+	}
+
+	sys_notify_init_callback(&cli.notify, dvfs_notify_cb);
+
+	if (res < 0) {
+		handle_dvfs_error(res);
```

Comment:
Error codes from the clock control api are ALL the same as in case of the dvfs? I've seen the ETIMEDOUT in the driver also beside that. Should we handle it additionally?

Reply:
We can handle it additionally but should we then retry? For now if it happened we enter error state.

Reply:
we can try to retry on ETIMEDOUT i think